src/Filters/remez_fir.jl

```diff
@@ -0,0 +1,829 @@
+    x = zeros(Float64, dimsize)
+    y = zeros(Float64, dimsize)
+
+    for j = 1 : nfcns
+        iext[j] = Int64(floor((j-1)*temp)) + 1
+    end
+    iext[nfcns+1] = ngrid
+
+    dev = 0.0     # deviation from the desired function, 
+                  # that is, the amount of "ripple" on the extremal set
+    devl = -1.0   # deviation on last iteration
+    nz  = nfcns+1
+    nzz = nfcns+2
+    niter = 0
+    ad = zeros(Float64, nz)
+    
+    jet = ((nfcns-1) ÷ 15) + 1
+
+    while true
+    
+        #
+        # Start next iteration
+        #        
+      @label L100
+        iext[nzz] = ngrid + 1
+        niter += 1
+        if niter > maxiter
+            warn("remez() iteration count exceeds maxiter = $maxiter, filter is not converged; try increasing maxiter")
+            # the filter is returned in its current, unconverged state.
+            break
+        end
+        
+        x[1:nz] = cospi.(2grid[iext[1:nz]])
+        
+        for j = 1 : nz
+            ad[j] = lagrange_interp(j, nz, jet, x)
+        end
+
+        dnum = 0.0
+        dden = 0.0
+        k = 1
+        for j = 1 : nz
+            l = iext[j]
+            dnum += ad[j] * des[l]
+            dden += k * ad[j] / wt[l]
+            k = -k
+        end
+        dev = dnum / dden
+        #printfmtln("  DEVIATION = {} at niter {}", dev, niter)
+
+        fill!(y, 0.0)
+        nu, dev = initialize_y(dev, nz, iext, des, wt, y)
+        
+        if dev <= devl
+            # finished
+            #printfmtln("dev {} <= devl {}, throwing", dev, devl)
+            throw(ErrorException("remez() - failure to converge at iteration $niter, try reducing transition band width"))
+        end
+        devl = dev
+
+        #
+        # SEARCH FOR THE EXTREMAL FREQUENCIES OF THE BEST APPROXIMATION
+        #
+    
+        # Between here and L370, the extremal index set is updated in a loop
+        # roughly over the index "j" - although the logic is complicated as
+        # the extremal set may grow or shrink in an iteration.
+        # j - the index of the current extremal being updated
+        # nz - the number of cosines in the approximation (including the constant term).
+        #      nz = nfcns + 1 where nfcns = nfilt / 2, and 
+        #      nfilt is the filter length or number of taps. 
+        #      For example, for a length 15 filter, nfcns = 7 and nz = 8. 
+        # jchgne - the number of extremal indices that changed this iteration
+        jchnge = 0
+        k1 = iext[1]
+        knz = iext[nz]
+        klow = 0
+        nut = -nu
+        j = 1
+
+      @label L200
+        j == nzz && (ynz = comp)   # equivalent to "if (j == nzz) ynz = comp; end"
+        j >= nzz && @goto L300
+        kup = iext[j+1]
+        l = iext[j]+1
+        nut = -nut
+        j == 2 && (y1 = comp)
+        comp = dev
+        l >= kup && @goto L220
+        err = (freq_eval(l,nz,grid,x,y,ad)-des[l]) * wt[l]
+        (nut*err-comp) <= 0.0 && @goto L220
```

Comment:
I'd prefer `nut*err <= comp` for the condition here (and for the following occurrences).